Cargo.toml

```diff
@@ -36,3 +40,80 @@ gtk_3_16 = ["gtk_3_10", "gtk/v3_16"]
 gtk_3_18 = ["gtk_3_16", "gtk/v3_18"] #for CI tools
 gtk_3_20 = ["gtk_3_18", "gtk/v3_20"] #for CI tools
 gtk_3_22 = ["gtk_3_20", "gtk/v3_22"] #for CI tools
+futures = ["futures-core", "futures-executor", "futures-channel", "futures-util", "glib/futures", "gio/futures"]
+
+[patch.'https://github.com/gtk-rs/gio']
+gio = { git = 'https://github.com/sdroege/gio-rs', branch = 'futures' }
+
+[[bin]]
+name = "basic"
+
+[[bin]]
+name = "builder_basics"
+
+[[bin]]
+name = "cairo_png"
+
+[[bin]]
+name = "cairo_threads"
+
+[[bin]]
+name = "cairotest"
+
+[[bin]]
+name = "child-properties"
+
+[[bin]]
+name = "clock"
+
+[[bin]]
+name = "drag_and_drop"
+
+[[bin]]
+name = "drag_and_drop_textview"
+
+[[bin]]
+name = "gio_futures"
+required-features = ["futures"]
+
+[[bin]]
+name = "grid"
+
+[[bin]]
+name = "gtktest"
+
+[[bin]]
+name = "menu_bar"
+
+[[bin]]
+name = "menu_bar_system"
+
+[[bin]]
+name = "multi_windows"
+
+[[bin]]
+name = "multithreading_context"
+
+[[bin]]
+name = "notebook"
+
+[[bin]]
+name = "pango_attributes"
+
+[[bin]]
+name = "simple_treeview"
+
+[[bin]]
+name = "sync_widgets"
+
+[[bin]]
+name = "text_viewer"
+
+[[bin]]
+name = "transparent_main_window"
+
+[[bin]]
+name = "treeview"
+
+[[bin]]
```

Comment:
Strange, why [[bin]] added?

Reply:
Oh, understand

Reply:
Maybe better do same as we do with version?
https://github.com/gtk-rs/examples/blob/master/src/bin/builder_basics.rs#L76-L85

Reply:
Why do in code what cargo can do for us? :) I would rather convert all the examples

Reply:
Current version don't accepted by cargo as it don't have `path`.

Reply:
Oops, it just because `variant.rs` do not exists

Reply:
Hmm I have that file :) I'll check and fix later